File: 1-two-sum/1-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        h = {}
        for i,n in enumerate(nums):
            need = target - n
            if need in h:
                return [h[need], i]
            h[n] = i

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> seen = new HashMap<>();
        
        for (int i = 0; i < nums.length;i++){
            int currentNum = nums[i];
            int need = target - currentNum;
            
            if(seen.containsKey(need)){
                return new int []{seen.get(need),i};
            }
            seen.put(currentNum, i);
        }
        return new int[]{};
    }
    
}